### PR TITLE
Fix config.ts: read settings and not use internal-build-dir by default

### PR DESCRIPTION
**Pull request by jonaprieto**
Before this PR and after 0.2.0, the settings window seems to no longer detect the options set for some unknown reason. It's possible that the logic used in the 'VsCodeSettings' file has become outdated due to the latest updates and the use of a very new vscode engine (+1.90.0). Despite my attempts to fix its behavior, I ultimately gave up. This PR now reads options according to the latest API. Additionally, the internal build directory is no longer used by default; it's now optional.

This PR goes on top:

- #131

Note: If the user changes any option in the Settings panel, they need to reload the window or reset the vscode for the changes to take effect.

### CODE REVIEW

File: src/tasks.ts [hunk 1]
```diff
@@ -236,7 +235,8 @@
       exec = new vscode.ShellExecution(JuvixExec + `dependencies update`);
       break;
     default:
-      exec = new vscode.ShellExecution(JuvixExec + `  ${input}`);
+      let cmd = (JuvixExec + ` ${input}`).replace('\n', '').replace('  ', ' ');
```
[the remark this answers is not in the document]
Comment (lukaszcz) [reply src/tasks.ts]: Is this a problem?

File: src/tasks.ts [hunk 2]
```diff
@@ -195,15 +195,20 @@ export async function JuvixTask(
   const config = new user.JuvixConfig();
   const JuvixExec = [config.getJuvixExec(), config.getGlobalFlags()].join(' ');
   let exec: vscode.ProcessExecution | vscode.ShellExecution | undefined;
-  const buildDir = config.getInternalBuildDir();
   const fl = args.slice(1).join(' ').trim();
   switch (name) {
     case 'run':
+      if (config.useInternalBuildDirOption()) {
+      const buildDir = config.getInternalBuildDir();
```
Comment (lukaszcz) [inline src/tasks.ts]: It seems like indentation is missing here

File: src/tasks.ts [hunk 3]
```diff
@@ -195,15 +195,20 @@ export async function JuvixTask(
...
       exec = new vscode.ShellExecution(
         JuvixExec +
         ` compile native --output ${buildDir}\${pathSeparator}out ${fl} && ${buildDir}\${pathSeparator}out`,
         { cwd: buildDir },
       );
+    } else {
+      exec = new vscode.ShellExecution(JuvixExec
+         + ` compile native --output ${fl}.out ${fl} && ${fl}.out`);
+    }
       break;
```
Comment (lukaszcz) [inline src/tasks.ts:213]: Again, seems like indentation is wrong

**Review by lukaszcz — CHANGES_REQUESTED** [review]
Fix indentation issues

File: src/tasks.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonaprieto) [reply src/tasks.ts]: Not, I removed this, I was using this for debugging purposes.

File: src/tasks.ts (hunk 3 above)
Comment (jonaprieto) [reply src/tasks.ts:213]: Fixed it.

File: src/tasks.ts (hunk 2 above)
Comment (jonaprieto) [reply src/tasks.ts]: Fixed it.

File: src/tasks.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lukaszcz) [reply src/tasks.ts]: This line is still in the code. So it's fine to replace only a single occurrence of '\n', and this is for debugging only?

**Review by lukaszcz — APPROVED** [review]

**Pull request merged**